androidbrowserhelper/src/main/java/com/google/androidbrowserhelper/trusted/LauncherActivity.java

```diff
@@ -172,6 +179,30 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
 
         new TwaSharedPreferencesManager(this)
                 .writeLastLaunchedProviderPackageName(mTwaLauncher.getProviderPackage());
+
+        mPackageManager = getPackageManager();
+        // Check whether ManageDataLauncherActivity is present in manifest.
+        Intent siteSettingsIntent = new Intent(this, ManageDataLauncherActivity.class);
+        siteSettingsIntent.setAction(ManageDataLauncherActivity.ACTION_MANAGE_TRUSTED_WEB_ACTIVITY_DATA);
+        List<ResolveInfo> activities = mPackageManager.queryIntentActivities(siteSettingsIntent,
+                PackageManager.MATCH_DEFAULT_ONLY);
+        boolean siteSettingsAvailable = activities.size() > 0;
+
+        // Check whether TWA provider package supports site settings.
+        String packageName = mTwaLauncher.getProviderPackage();
+        boolean supportsTWASiteSettings = ManageDataLauncherActivity.packageSupportsSiteSettings(packageName, mPackageManager);
+
+        if(siteSettingsAvailable && supportsTWASiteSettings) {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N_MR1) {
```

Comment:
The version check should probably go right at the top of the function - if we can't set shortcuts, there's no reason to have checked all the stuff before.